libraries/AP_Mount/AP_Mount_XFRobot.cpp

```diff
@@ -0,0 +1,594 @@
+    }
+
+    // check for failure to start/stop recording
+    if (recv_packet.content.main.order == FunctionOrder::RECORD_VIDEO) {
+        if (recv_packet.content.param.execution_state == 0x0) {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s recording %s", send_text_prefix, recording.request_start ? "ON" : "OFF");
+        } else {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s failed to %s recording", send_text_prefix, recording.request_start ? "start" : "stop");
+        }
+        recording.request_ms = 0;
+    }
+}
+
+// send_target_angles
+void AP_Mount_XFRobot::send_target_angles(const MountTarget& angle_target_rad)
+{
+    // exit immediately if not initialised or not enough space to send packet
+    if (!_initialised || _uart->txspace() < sizeof(SetAttitudePacket)) {
+        return;
+    }
+
+    // prepare packet to send to gimbal
+    SetAttitudePacket set_attitude_packet {};
+
+    // bytes 0~1: headers
+    set_attitude_packet.content.main.header1 = SEND_HEADER1;
+    set_attitude_packet.content.main.header2 = SEND_HEADER2;
+
+    // byte 2~3: length (uint16)
+    set_attitude_packet.content.main.length = htole16(sizeof(SetAttitudePacket));
+
+    // byte 4: protocol version (0x02)
+    set_attitude_packet.content.main.version = PROTOCOL_VERSION;
+
+    // byte 5~6: roll control value (int16, -18000 ~ +18000)
+    // byte 7~8: pitch control value (int16, -18000 ~ +18000)
+    // byte 9~10: yaw control value (int16, -18000 ~ +18000)
+    set_attitude_packet.content.main.roll_control = htole16(constrain_int16(degrees(angle_target_rad.roll) * 100, -18000, 18000));
+    set_attitude_packet.content.main.pitch_control = htole16(constrain_int16(degrees(angle_target_rad.pitch) * 100, -9000, 9000));
+    set_attitude_packet.content.main.yaw_control = htole16(constrain_int16(degrees(angle_target_rad.get_bf_yaw()) * 100, -18000, 18000));
+
+    // byte 11: status, Bit0:INS valid, Bit2:control values valid
+    const uint8_t status_ins_bit = AP::ahrs().have_inertial_nav() ? (1 << 0) : 0;
+    set_attitude_packet.content.main.status = status_ins_bit | (1 << 2);
+
+    // byte 12~13: absolute roll angle of vehicle (int16, -18000 ~ +18000)
+    // byte 14~15: absolute pitch angle of vehicle (int16, -9000 ~ +9000)
+    // byte 16~17: absolute yaw angle of vehicle (uint16, 0 ~ 36000)
+    set_attitude_packet.content.main.roll_abs = htole16(constrain_int16(degrees(AP::ahrs().get_roll()) * 100, -18000, 18000));
+    set_attitude_packet.content.main.pitch_abs = htole16(constrain_int16(degrees(AP::ahrs().get_pitch()) * 100, -9000, 9000));
+    set_attitude_packet.content.main.yaw_abs = htole16(constrain_int16(degrees(wrap_PI(AP::ahrs().get_yaw())) * 100, -18000, 18000));
+
+    // byte 18~19: North acceleration of vehicle (int16, cm/s/s)
+    // byte 20~21: East acceleration of vehicle (int16, cm/s/s)
+    // byte 22~23: Upward acceleration of vehicle (int16, cm/s/s)
+    Vector3f accel_ef = AP::ahrs().get_accel_ef();
```

Comment:
```suggestion
    const Vector3f &accel_ef = AP::ahrs().get_accel_ef();
```

Reply:
thanks for the review, changed to const reference